src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -256,7 +255,12 @@ protected function resolveExprConstFetch(Expr\ConstFetch $node)
 
     protected function resolveExprClassConstFetch(Expr\ClassConstFetch $node)
     {
-        $refClass     = $this->fetchReflectionClass($node->class);
+        $classToReflect = $node->class;
```

Comment:
> Yes, I know you can do a constant fetch from a class name stored in a variable. I'm unsure what other Exprs are valid. 

I've meant `Example #4 Constant expression example` from http://php.net/manual/en/language.oop5.constants.php page:

```php
const ONE = 1;

class foo {
    // As of PHP 5.6.0
    const TWO = ONE * 2;
    const THREE = ONE + self::TWO;
    const SENTENCE = 'The value of THREE is '.self::THREE;
}
```

Working since PHP 5.6 only. This way you can create test for that kind of syntax.

> As far as adding corresponding test cases, I agree, but I was lumping my test changes into a separate PR. 

The tests for changed code needs to come as part of PR, where that code is changed. Later you can send another PR for improving code coverage.